### PR TITLE
Add Table index idea

### PR DESCRIPTION
**Pull request by taldcroft**
Add GSOC 2015 idea for indexed columns in a Table object.

@Cadair

### CODE REVIEW

File: gsoc2015/ideas_astropy.md [hunk 1]
```diff
@@ -132,3 +132,35 @@ motion. This project could be:
...
+core astropy class for storing and manipulating tabular data.  Currently it
+supports a limited set of database-like capabilities including table joins and
+grouping.  A natural extension of this is to provide the ability to create and
+maintain an index on one or more columns as well as a table primary key.  With
+these indexed columns available then certain selection and query operations
+could be highly optimized.  The challenge is to maintain the integrity of the
+indexes as column or table properties change, using state of the art algorithms
+for high performance.
+
+To summarize:
```
Comment (dpshelio) [inline gsoc2015/ideas_astropy.md]: `#### Expected Outcomes` to keep it in line with other ideas.

**Comment by Cadair** [comment]
@taldcroft looks good! I wonder if it would be worth adding a paragraph about specific types of index columns, such as time series etc. to give the students an idea of the use cases?

**Comment by taldcroft** [comment]
@Cadair - good idea.  Can you give me such a paragraph and I'll toss it in?

**Comment by Cadair** [comment]
@taldcroft I will try and cook one up in a few, I am just writing the student application guidelines at the moment.

**Comment by taldcroft** [comment]
@astrofrog - @Cadair and myself are good with this now if you want to merge.

**Pull request merged**